### PR TITLE
AUT-2816: Align OTP TTL with Count TTL where OTP TTL > Count TTL

### PR DESCRIPTION
**Pull request by LazarAlexandru-Constantin**
## What

Align OTP TTL with Count TTL.
As proposed in the description of [AUT-2816](https://govukverify.atlassian.net/browse/AUT-2816), when the count TTL is smaller than the OTP TTL, a user is able to send more than 5 attempts for one code. This PR is looking to fix that by introducing a new variable for the count TTL which is only used in the account creation journeys. 

For scenario 2, in build, the Default_Otp_Code_Expiry was not set in the tf file, making the ResetPasswordRequestHandler.tf use the standard 900 second value for the Otp Code Expiry. This would make the OTP TTL > Count TTL.

## How to review

1. Code Review

[AUT-2816]: https://govukverify.atlassian.net/browse/AUT-2816?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: frontend-api/src/test/java/uk/gov/di/authentication/frontendapi/validation/AuthAppCodeProcessorTest.java [hunk 1]
```diff
@@ -159,6 +160,23 @@ void returnsCorrectErrorWhenAuthCodeIsInvalid(JourneyType journeyType, String au
         assertEquals(Optional.of(ErrorResponse.ERROR_1043), authAppCodeProcessor.validateCode());
     }
 
+    @Test
+    void IncrementsWithCorrectTTLWhenAuthCodeIsInvalidInCreationJourney() {
```
Comment (ayoshebby) [inline frontend-api/src/test/java/uk/gov/di/authentication/frontendapi/validation/AuthAppCodeProcessorTest.java:164]: I suggest camelCase the name of this test to `incrementsWithCorrectTTLWhenAuthCodeIsInvalidInCreationJourney`

Comment (LazarAlexandru-Constantin) [reply frontend-api/src/test/java/uk/gov/di/authentication/frontendapi/validation/AuthAppCodeProcessorTest.java:164]: Nice spot, agreed!

**Review by ayoshebby — APPROVED** [review]

File: frontend-api/src/main/java/uk/gov/di/authentication/frontendapi/validation/AuthAppCodeProcessor.java [hunk 2]
```diff
@@ -75,7 +77,12 @@ public Optional<ErrorResponse> validateCode() {
             return Optional.of(ErrorResponse.ERROR_1042);
         }
 
-        incrementRetryCount(MFAMethodType.AUTH_APP);
+        if (!nonRegistrationJourneyTypes.contains(codeRequest.getJourneyType())
```
Comment (dbes-gds) [inline frontend-api/src/main/java/uk/gov/di/authentication/frontendapi/validation/AuthAppCodeProcessor.java:80]: just want to review this before it's merged

**Comment by CraigRyder** [comment]
Merging on behalf of Alex as he doesn't have permissions to do so

File: shared/src/main/java/uk/gov/di/authentication/shared/helpers/ValidationHelper.java [hunk 3]
```diff
@@ -154,9 +157,13 @@ public static Optional<ErrorResponse> validateVerificationCode(
             return Optional.of(ErrorResponse.ERROR_1002);
         }
 
-        codeStorageService.increaseIncorrectMfaCodeAttemptsCount(emailAddress);
+        if (configurationService.supportAccountCreationTTL()
+                && List.of(VERIFY_EMAIL, VERIFY_PHONE_NUMBER).contains(notificationType)) {
+            codeStorageService.increaseIncorrectMfaCodeAttemptsCountAccountCreation(emailAddress);
+        } else codeStorageService.increaseIncorrectMfaCodeAttemptsCount(emailAddress);
```
Comment (dbes-gds) [inline shared/src/main/java/uk/gov/di/authentication/shared/helpers/ValidationHelper.java:163]: missing brackets?

File: shared/src/main/java/uk/gov/di/authentication/shared/helpers/ValidationHelper.java [hunk 4]
```diff
@@ -154,9 +157,13 @@ public static Optional<ErrorResponse> validateVerificationCode(
             return Optional.of(ErrorResponse.ERROR_1002);
         }
 
-        codeStorageService.increaseIncorrectMfaCodeAttemptsCount(emailAddress);
+        if (configurationService.supportAccountCreationTTL()
+                && List.of(VERIFY_EMAIL, VERIFY_PHONE_NUMBER).contains(notificationType)) {
```
Comment (dbes-gds) [inline shared/src/main/java/uk/gov/di/authentication/shared/helpers/ValidationHelper.java:161]: Sure that phone number should be included here?  Also does VERIFY_EMAIL get used in scenarios other than account creation (account recovery?)

File: shared/src/main/java/uk/gov/di/authentication/shared/services/CodeStorageService.java [hunk 5]
```diff
@@ -67,11 +67,24 @@ public void increaseIncorrectMfaCodeAttemptsCount(String email) {
                 configurationService.getLockoutCountTTL());
     }
 
+    public void increaseIncorrectMfaCodeAttemptsCountAccountCreation(String email) {
```
Comment (dbes-gds) [inline shared/src/main/java/uk/gov/di/authentication/shared/services/CodeStorageService.java:70]: will this clash with the other count as they have the same prefix?

File: frontend-api/src/main/java/uk/gov/di/authentication/frontendapi/validation/AuthAppCodeProcessor.java (hunk 2 above)
Comment (dbes-gds) [reply frontend-api/src/main/java/uk/gov/di/authentication/frontendapi/validation/AuthAppCodeProcessor.java:80]: the changes should only apply to the account creation email otp, not the 2nd factor

**Review by dbes-gds — APPROVED** [review]

**Pull request merged**